### PR TITLE
Create bug-fix release 0.23.1

### PR DESCRIPTION
**Pull request by maliasadi**
**Context:**
- Currently, `lightning.gpu` doesn't support finite shots. This PR adds a clear error message for any tries with `shots != None`.  
- `make format` (and `make format-cpp`) now accepts the version number in case `clang-format` is not installed or not satisfying the default version -- that is currently `12`. This update avoids inconsistencies between different clang-format aliases and eases trying different versions when needed.

### CODE REVIEW

File: pennylane_lightning_gpu/lightning_gpu.py [hunk 1]
```diff
@@ -117,6 +117,9 @@ class LightningGPU(LightningQubit):
     }
 
     def __init__(self, wires, *, shots=None, sync=True):
+        if shots is not None:
```
Comment (mlxd) [inline pennylane_lightning_gpu/lightning_gpu.py:120]:
```suggestion
        if shots:
```

should allow `shots=0` and `None` to skip this.

File: tests/test_apply.py [hunk 2]
```diff
@@ -569,6 +569,14 @@ def test_load_default_qubit_device(self):
...
         assert dev.short_name == "lightning.gpu"
 
+    def test_with_shots(self):
+        """Test that lightning.gpu does not support finite shots"""
+
+        with pytest.raises(
+            ValueError, match="lightning.gpu does not support finite shots, please use shots=None"
+        ):
+            qml.device("lightning.gpu", wires=2, shots=1)
+
```
Comment (mlxd) [inline tests/test_apply.py:579]: Adding a test for `shots=0` and `shots=None` too and checking if no exceptions raised should be good.

**Review by mlxd — CHANGES_REQUESTED** [review]
Nice work @maliasadi 
2 small suggestions, and I think we can call this done.

File: bin/format [hunk 3]
```diff
@@ -28,7 +28,7 @@ def parse_version(version_string):
     return int(m.group(1))
 
 def check_bin():
-    command = f"clang-format"
+    command = f"clang-format-{DEFAULT_CLANG_FORMAT_VERSION}"
```
Comment (maliasadi) [inline bin/format:31]: Hi @mlxd. `make format` doesn't work without a default version number on my machines. 
This is also compatible with the one in Lightning. What is your call?

File: tests/test_apply.py [hunk 4]
```diff
@@ -633,26 +649,6 @@ def circuit(x):
 
         assert np.isclose(circuit(p), 1, atol=tol, rtol=0)
 
-    def test_nonzero_shots(self, tol):
```
Comment (maliasadi) [inline tests/test_apply.py:636]: I removed this as LightningGPU doesn't currently support `shots > 0`.

Comment (mlxd) [reply tests/test_apply.py:636]: I think this is fine, but keeping it as an expected failure for now. When we get support for `shots>0` that test will let us know an update is needed.

File: bin/format (hunk 3 above)
Comment (maliasadi) [reply bin/format]: Thanks for the suggestion. Just update it!

**Review by mlxd — APPROVED** [review]

**Comment by maliasadi** [comment]
> @maliasadi With the shots support added as of [1e9ea1c](https://github.com/PennyLaneAI/pennylane-lightning-gpu/commit/1e9ea1c20b91a3fe0001833c561e89e6ff563f83), do we need to merge any of the features from this PR/branch into the main branch before the next release, or is this still existing for the 0.23.1 release?

Hi @mlxd. The main branch is already updated w.r.t to the PR. I think that we no longer need this one and I can close it.
